amethyst_network/src/metrics/console_metrics.rs

```diff
@@ -0,0 +1,36 @@
+use super::{MetricObserver, NetworkMetricObject};
+use serde::Serialize;
+use std::fs::File;
+use std::time::{Duration, Instant};
+
+/// This will handle changing metrics and write them to the console.
+/// Console write interval could be managed by passing in a `Duration` at creation time.
+pub struct ConsoleMetrics {
+    // this will be used for the writing interval to the console.
+    last_send: Instant,
+    // interval rate
+    update_rate: Duration,
+}
+
+impl ConsoleMetrics {
+    /// Create a new instance of `ConsoleMetrics` by passing in a interval at which the metrics will be written to screen.
+    pub fn new(update_rate: Duration) -> ConsoleMetrics {
+        ConsoleMetrics {
+            last_send: Instant::now(),
```

Comment:
This should probably be an option to represent this properly since, technically you haven't sent anything so how can the last_send be now?  😄

Reply:
I was thinking about that. However, if I use options I need to unwrap the option which I don't like since last_seen will always be used. So I figured just having the time instance there to initialize it would be fine.

Reply:
You can do 
```rust
if let Some(last_send) = self.last_send {
	if last_send.elapsed() >= self.update_rate { ... }
} else {
	// send here
}

Reply:
Yea, I know how that work. But it is more about the logic needed to unwrap the option. Since last seen is not really optional but not known at initializing time.